tekton/ci/custom-tasks/pr-status-updater/config/500-controller.yaml

```diff
@@ -0,0 +1,87 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: pr-status-updater-controller
+  namespace: tekton-pipelines
+  labels:
+    app.kubernetes.io/name: controller
+    app.kubernetes.io/component: controller
+    app.kubernetes.io/instance: default
+    app.kubernetes.io/version: "devel"
+    app.kubernetes.io/part-of: pr-status-updater
+    # tekton.dev/release value replaced with inputs.params.versionTag in pipeline/tekton/publish.yaml
+    pipeline.tekton.dev/release: "devel"
+    # labels below are related to istio and should not be used for resource lookup
+    version: "devel"
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: controller
+      app.kubernetes.io/component: controller
+      app.kubernetes.io/instance: default
+      app.kubernetes.io/part-of: pr-status-updater
+  template:
+    metadata:
+      annotations:
+        cluster-autoscaler.kubernetes.io/safe-to-evict: "false"
+      labels:
+        app.kubernetes.io/name: controller
+        app.kubernetes.io/component: controller
+        app.kubernetes.io/instance: default
+        app.kubernetes.io/version: "devel"
+        app.kubernetes.io/part-of: pr-status-updater
+        # tekton.dev/release value replaced with inputs.params.versionTag in pipeline/tekton/publish.yaml
+        pipeline.tekton.dev/release: "devel"
+        # labels below are related to istio and should not be used for resource lookup
+        app: pr-status-updater-controller
+        version: "devel"
+    spec:
+      serviceAccountName: pr-status-updater-controller
+      containers:
+        - name: pr-status-updater-custom-task
+          image: ko://github.com/tektoncd/plumbing/tekton/ci/custom-tasks/pr-status-updater/cmd/pr-status-updater
+          securityContext:
+            allowPrivilegeEscalation: false
+            # User 65532 is the distroless nonroot user ID
+            runAsUser: 65532
+          env:
+            - name: SYSTEM_NAMESPACE
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.namespace
+            - name: CONFIG_LEADERELECTION_NAME
+              value: config-leader-election
+            - name: CONFIG_LOGGING_NAME
+              value: config-logging
+            - name: CONFIG_OBSERVABILITY_NAME
+              value: config-observability
+            - name: METRICS_DOMAIN
+              value: custom.tekton.dev/pr-status-updater
+            - name: GIT_KIND
+              value: github
+            - name: GIT_SERVER
+              value: https://github.com
+            - name: GIT_OWNER
+              value: tektoncd
+            - name: GIT_USER
+              value: tekton-robot
+            - name: "GIT_TOKEN"
+              valueFrom:
+                secretKeyRef:
+                  name: bot-token-github
+                  key: bot-token
```

Comment:
Ideally, we would have the name of the secret here instead, but as long as this is used in dogfooding only it's ok.

Reply:
Yeah, that was my thinking.